src/components/NotesMindMap/NotesMindMapReducer.js

```diff
@@ -99,9 +91,16 @@ export const notesMindMapReducer = (state = defaultState, { type, data }) => {
     return newState;
   }
 
+  const addRootToGraph = () => {
+    const newState = { ...state };
+    newState.rootNote = data;
+    newState.notes.push({ id: data.id, label: data.name });
```

Comment:
Isn't `push` mutating the state?

https://redux.js.org/recipes/structuring-reducers/immutable-update-patterns/#inserting-and-removing-items-in-arrays

Reply:
My bad :( Removed